net/golioth/Kconfig

```diff
@@ -26,6 +25,51 @@ module = GOLIOTH
 module-str = golioth
 source "subsys/logging/Kconfig.template.log_config"
 
+choice GOLIOTH_AUTH_METHOD
+	prompt "Authentication method support"
+
+config GOLIOTH_AUTH_METHOD_PSK
+	bool "PSK-ID / PSK only"
+	imply GOLIOTH_AUTH_PSK_MBEDTLS_DEPS
+	help
+	  Only PSK based authentication will be supported.
+
+config GOLIOTH_AUTH_METHOD_CERT
+	bool "Certificate based only"
+	imply GOLIOTH_AUTH_CERT_MBEDTLS_DEPS
+	help
+	  Only certificate based authentication will be supported.
+
+endchoice
+
+config GOLIOTH_AUTH_PSK_MBEDTLS_DEPS
+	bool "mbedTLS dependencies for PSK auth"
+	depends on MBEDTLS_BUILTIN
+	select MBEDTLS_KEY_EXCHANGE_PSK_ENABLED
+	help
+	  Select mbedTLS dependencies for PSK based authentication.
+
+	  This is a convenience option when using mbedTLS as a library to make
+	  (D)TLS connection with Golioth cloud.
+
+config GOLIOTH_AUTH_CERT_MBEDTLS_DEPS
+	bool "mbedTLS dependencies for certificate based auth"
+	depends on MBEDTLS_BUILTIN
+	# At least one ECP needs to be selected. This was chosen arbitrarily.
+	imply MBEDTLS_ECP_DP_SECP256R1_ENABLED if MBEDTLS_BUILTIN
+	select MBEDTLS_KEY_EXCHANGE_ECDHE_ECDSA_ENABLED if MBEDTLS_BUILTIN
+	help
+	  Select mbedTLS dependencies for certificate based authentication.
+
+	  This is a convenience option when using mbedTLS as a library to make
+	  (D)TLS connection with Golioth cloud.
+
+config GOLIOTH_HOSTNAME_VERIFICATION
+	bool "Hostname verification"
+	default y if GOLIOTH_AUTH_METHOD_CERT || GOLIOTH_AUTH_METHOD_PSK_AND_CERT
```

Comment:
`GOLIOTH_AUTH_METHOD_PSK_AND_CERT`

I don't see this anywhere else.

Reply:
Good catch! That is a leftover from refactoring. Initially I wanted to have 3rd option of "psk+cert" at the same time with possibility to choose at runtime, e.g. depending on provisioned credentials or so. But it was too much effort compared to the value IMO. Maybe we could introduce such option in the future, but first it would make more sense to refactor (split) TLS credentials management.